mule-extensions-api/src/main/java/org/mule/runtime/extension/api/component/value/ObjectValueDeclarer.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.component.value;
+
+import org.mule.api.annotation.NoImplement;
+
+import java.util.function.Consumer;
+
+/**
+ * Describes an Object value to be created.
+ *
+ * It describes the Object structure but not its format. This should be used the same way regardless of the final format it will
+ * have, be it a JSON document or a CSV line.
```

Comment:
`This should be used the same way regardless of its actual format (Java object, Json/XML document, CSV line, etc).`